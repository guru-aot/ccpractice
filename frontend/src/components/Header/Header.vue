/**
 * Header of app
 */

<template>
  <v-app-bar app>
    <v-toolbar-title class="headline text-uppercase">
      <span>Home</span>
    </v-toolbar-title>
    <v-spacer></v-spacer>
    <v-btn text to="/author" link v-if="isAdmin">
      <span class="mr-2">Author</span>
    </v-btn>
    <v-btn text to="/" link>
      <span class="mr-2">Home</span>
    </v-btn>
    <v-btn text to="/login" link v-if="!isLoggedin">
      <span class="mr-2">Login</span>
    </v-btn>
    <v-menu open-on-hover top offset-y v-if="isLoggedin">
      <template v-slot:activator="{ on }">
        <v-avatar color="indigo" v-on="on">
          <v-icon dark>mdi-account-circle</v-icon>
        </v-avatar>
      </template>

      <v-list>
        <v-list-item v-if="isUser">
          <v-list-item-title>
            <v-btn text to="/profile" link>
              <span class="mr-2">{{userProfile.lastName}}</span>
            </v-btn>
          </v-list-item-title>
        </v-list-item>
        <v-list-item @click="logout">
          <v-list-item-title>
            <v-btn text v-if="isLoggedin">
              <span class="mr-2">Logout</span>
            </v-btn>
          </v-list-item-title>
        </v-list-item>
      </v-list>
    </v-menu>
  </v-app-bar>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import { Getter, namespace } from 'vuex-class';
const KeyCloakModule = namespace('KeyCloakModule');

@Component
export default class Header extends Vue {
  @KeyCloakModule.Getter('userProfile') private userProfile!: [];
  @KeyCloakModule.Getter('isLoggedin') private isLoggedin!: boolean;
  @KeyCloakModule.Getter('isUser') private isUser!: boolean;
  @KeyCloakModule.Getter('isAdmin') private isAdmin!: boolean;

  /**
   * make user logout
   * @returns {} dispatch
   */
  private logout() {
    this.$store.dispatch('KeyCloakModule/setLogout');
  }
}
</script>
