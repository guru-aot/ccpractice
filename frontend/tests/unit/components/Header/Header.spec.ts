import { shallowMount, createLocalVue } from '@vue/test-utils';
import Header from '@/components/Header/Header.vue';
import Vuex from 'vuex';
import vuetify from 'vuetify';

describe('Header.vue', () => {
  let wrapper: any;
  beforeEach(() => {
    const localVue = createLocalVue();
    localVue.use(vuetify);
    localVue.use(Vuex);
    const store = new Vuex.Store({
      modules: {
        KeyCloakModule: {
          namespaced: true,
          state: { isLoggedin: false },
          getters: {
            userProfile: jest.fn(),
            isLoggedin: jest.fn(),
          },
        },
      },
    });
    wrapper = shallowMount(Header, {
      localVue,
      store,
    });
  });

  it('renders props when passed', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
