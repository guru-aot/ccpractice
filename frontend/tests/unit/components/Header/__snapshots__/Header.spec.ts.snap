// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.vue renders props when passed 1`] = `
<v-app-bar-stub
  app="true"
  extensionheight="48"
  src=""
  tag="header"
  tile="true"
  value="true"
>
  <v-toolbar-title-stub
    class="headline text-uppercase"
  >
    <span>
      Home
    </span>
  </v-toolbar-title-stub>
   
  <v-spacer-stub />
   
  <!---->
   
  <v-btn-stub
    activeclass=""
    link="true"
    tag="button"
    text="true"
    to="/"
    type="button"
  >
    <span
      class="mr-2"
    >
      Home
    </span>
  </v-btn-stub>
   
  <v-btn-stub
    activeclass=""
    link="true"
    tag="button"
    text="true"
    to="/login"
    type="button"
  >
    <span
      class="mr-2"
    >
      Login
    </span>
  </v-btn-stub>
   
  <!---->
</v-app-bar-stub>
`;
